runtime/src/software_interrupt.c

```diff
@@ -72,9 +79,6 @@ sigalrm_propagate_workers(siginfo_t *signal_info)
 		for (int i = 0; i < runtime_worker_threads_count; i++) {
 			if (pthread_self() == runtime_worker_threads[i]) continue;
 
-			/* All threads should have been initialized */
-			assert(runtime_worker_threads[i] != 0);
```

Comment:
This assertion makes sense to me, but I would have it before line 73. If the assertion is on a value, we should trigger the assertion *before* we use the value.

Reply:
Okay, that makes sense. Putting it before that line.